### PR TITLE
Do not run benchmark tests in CI

### LINKED ISSUE #2659: We are running benchmark tests in CI
### Describe your environment

OS: (e.g, Ubuntu)
Python version: (e.g., Python 3.8.10)
Package version: (e.g., 0.46.0)

### What happened?

We are running benchmark tests in CI

### Steps to Reproduce

Run CI

### Expected Result

No benchmark tests being run

### Actual Result

Benchmark tests are being run

### Additional context

It makes no sense to run them in CI unless these tests can fail in a useful manner, for example if they detect a drop in performance, right now they are not configured to do so. Also, we should not run the benchmark tests in CI, it is useless because they need a self-hosted runner in order to not be affected by server load and give meaningful results.

### Would you like to implement a fix?

None

### PR DESCRIPTION
**Pull request by ocelotl**
Fixes #2659

### CODE REVIEW

**Comment by ocelotl** [comment]
Benchmark tests are still being run in CI, investigating...

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -34,7 +34,7 @@ jobs:
             ~/.cache/pip
           key: v7-misc-tox-cache-${{ matrix.tox-environment }}-${{ hashFiles('tox.ini', 'dev-requirements.txt', 'gen-requirements.txt', 'docs-requirements.txt') }}
       - name: run tox
-        run: tox -e ${{ matrix.tox-environment }}
+        run: tox -e ${{ matrix.tox-environment }}  --benchmark-skip
```
Comment (xrmx) [inline .github/workflows/test.yml:37]:
```suggestion
        run: tox -e ${{ matrix.tox-environment }}
```

**Review by lzchen — APPROVED** [review]

**Comment by lzchen** [comment]
@ocelotl 

Do we even have benchmark tests for the contrib repo?

**Review by xrmx — APPROVED** [review]

**Review by emdneto — APPROVED** [review]

**Comment by ocelotl** [comment]
> @ocelotl
> 
> Do we even have benchmark tests for the contrib repo?

Yes, [here](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/sdk-extension/opentelemetry-sdk-extension-aws/tests/performance/benchmarks/trace/test_benchmark_aws_xray_ids_generator.py) and [here](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/propagator/opentelemetry-propagator-aws-xray/tests/performance/benchmarks/test_benchmark_aws_xray_propagator.py).

**Pull request merged**